packages/utils/src/lib/reports/generate-md-report-categoy-section.ts

```diff
@@ -15,64 +15,76 @@ import {
+  const { isScoreListed = (_: number) => true } = options ?? {};
   const { categories, plugins } = report;
   return new MarkdownDocument().table(
     [
       { heading: '🏷 Category', alignment: 'left' },
       { heading: '⭐ Score', alignment: 'center' },
       { heading: '🛡 Audits', alignment: 'center' },
     ],
-    categories.map(({ title, refs, score, isBinary }) => [
-      // @TODO refactor `isBinary: boolean` to `targetScore: number` #713
-      // The heading "ID" is inferred from the heading text in Markdown.
-      md.link(`#${slugify(title)}`, title),
-      md`${scoreMarker(score)} ${md.bold(
-        formatReportScore(score),
-      )}${binaryIconSuffix(score, isBinary)}`,
-      countCategoryAudits(refs, plugins).toString(),
-    ]),
+    categories
+      .filter(({ score }) => isScoreListed(score))
+      .map(({ title, refs, score, isBinary }) => [
+        // @TODO refactor `isBinary: boolean` to `targetScore: number` #713
+        // The heading "ID" is inferred from the heading text in Markdown.
+        md.link(`#${slugify(title)}`, title),
+        md`${scoreMarker(score)} ${md.bold(
+          formatReportScore(score),
+        )}${binaryIconSuffix(score, isBinary)}`,
+        countCategoryAudits(refs, plugins).toString(),
+      ]),
   );
 }
 
 export function categoriesDetailsSection(
   report: Required<Pick<ScoredReport, 'plugins' | 'categories'>>,
+  options?: ScoreFilter,
 ): MarkdownDocument {
+  const { isScoreListed = (_: number) => true } = options ?? {};
   const { categories, plugins } = report;
 
   return new MarkdownDocument()
     .heading(HIERARCHY.level_2, '🏷 Categories')
-    .$foreach(categories, (doc, category) =>
-      doc
-        .heading(HIERARCHY.level_3, category.title)
-        .paragraph(metaDescription(category))
-        .paragraph(
-          md`${scoreMarker(category.score)} Score: ${md.bold(
-            formatReportScore(category.score),
-          )}${binaryIconSuffix(category.score, category.isBinary)}`,
-        )
-        .list(
-          category.refs.map(ref => {
-            // Add group details
-            if (ref.type === 'group') {
-              const group = getSortableGroupByRef(ref, plugins);
-              const groupAudits = group.refs.map(groupRef =>
-                getSortableAuditByRef(
-                  { ...groupRef, plugin: group.plugin, type: 'audit' },
-                  plugins,
-                ),
-              );
-              const pluginTitle = getPluginNameFromSlug(ref.plugin, plugins);
-              return categoryGroupItem(group, groupAudits, pluginTitle);
-            }
-            // Add audit details
-            else {
-              const audit = getSortableAuditByRef(ref, plugins);
-              const pluginTitle = getPluginNameFromSlug(ref.plugin, plugins);
-              return categoryRef(audit, pluginTitle);
-            }
-          }),
-        ),
+    .$foreach(
+      categories.filter(({ score }) => isScoreListed(score)),
```

Comment:
This pattern of destructuring with default "always" true implementation and then applying filter to `score` property is a bit repetitive.

If you were to create a helper function like this:

```ts
function scoreFilter<T extends { score: number }>(options: ScoreFilter | undefined) {
  const { isScoreListed = () => true } = options ?? {};
  return ({ score }: T) => isScoreListed(score);
}
```

Then the usage could be simplified to a one-liner:
- `categories.filter(scoreFilter(options))` (in `categoriesOverviewSection` and `categoriesDetailsSection` function),
- `plugin.audits.filter(scoreFilter(options))` (in `auditsSection` function).

Reply:
done